app/scripts/models/reliers/relier.js

```diff
@@ -109,6 +109,54 @@ define(function (require, exports, module) {
     },
 
     /**
+     * Set a value based on a value in window.location.search. Throws error
+     * if sourceName param is not in window.location.search.
+     *
+     * Throws Error mapped to `MISSING_PARAMETER` in Errors object.
+     *
+     * @param {string} sourceName - name of the search parameter
+     * @param {string} destName - name to set in model
+     * @param {Errors} Errors - corresponding Errors object
+     * @throws {error}
+     */
+    importRequiredSearchParam: function (sourceName, destName, Errors) {
+      var self = this;
+      self.importSearchParam(sourceName, destName);
+      if (! self.has(destName)) {
```

Comment:
If `self.has` is changed to `self.get`, then this is pretty much identical to the version in [reliers/oauth.js](https://github.com/mozilla/fxa-content-server/pull/3469/files#diff-ca4d7096469c2a76694bb8214760cbaaR187). We should update the `has` to `get` here and use this version everywhere.